src/ghostfragment/capping/DCLC.cpp

```diff
@@ -0,0 +1,128 @@
+    }
+    // If bonds exist, return their average length
+    if(existing_bonds > 0){
+        ave_length = bond_lengths/existing_bonds;
+    }
+    return ave_length;
+}
+
+namespace {
+constexpr auto module_desc = R"""(
+##################
+DCLC Capper
+##################
+
+This module closes off the valencies of the input fragments using single
+atoms. More specifically for each bond A-B, such that atom A is in the
+fragment, and atom B is not, an atom (default is a hydrogen atom) will be
+added to the fragment. The new atom will be placed at a distance determined
+by the average A-X bond in the molecule, where X is the capping atom. If no
+such bonds exist, then the bond distance is given by the length of the typical
+A-X bond (i.e. the sum of the atoms' respective covalent radii).
+
+The inputs to this module are fragments. In general these inputs are
+non-disjoint, for this reason we choose to establish connectivity at an
+atomic level.
+
+#. Generate atomic connectivity
+#. Determine caps we need
+#. Pair each fragment with its set of caps
+
+)""";
+} // end namespace
+
+MODULE_CTOR(DCLC) {
+    satisfies_property_type<my_pt>();
+
+    add_submodule<connect_pt>("Connectivity");
+
+    add_input<atom_type>("capping atom")
+      .set_description("atom to use as the cap")
+      .set_default(atom_type{"H", 1ul, 1837.289, 0.0, 0.0, 0.0});
+}
+
+MODULE_RUN(DCLC) {
+    const auto& [frags]    = my_pt::unwrap_inputs(inputs);
+    auto cap               = inputs.at("capping atom").value<atom_type>();
+    float bond_length      = 0;
+    float existing_bonds   = 0;
+    float original_bond    = 0;
+
+    // Step 1. Generate atomic connectivity
+    const auto& mol   = frags.supersystem();
+    const auto& conns = submods.at("Connectivity").run_as<connect_pt>(mol);
+
+    // Step 2. Make the caps
+    using result_type  = traits_t::result_type;
+    using cap_set_type = typename result_type::value_type;
+
+    result_type capped_frags; // Will be the result
+    for(const auto& frag_i : frags) {
+        cap_set_type caps_i; // Will be the set of caps for this fragment
+        for(const auto& atom_i : frag_i) {
+            // Loop over atoms bonded to atom_i
+            for(const auto& atom_j : conns.bonded_atoms(atom_i)) {
+                // Check if atom_j is in the fragment, if so no cap is needed
+                if(frag_i.count(mol[atom_j])) continue;
+
+                // Make the cap
+                atom_type new_cap(cap);
+                original_bond = (mol[atom_i].as_nucleus()
+                - mol[atom_j].as_nucleus()).magnitude();
+                for(atom_type::size_type i = 0; i < 3; ++i){
+                    new_cap.coord(i) = mol[atom_i].coord(i)
+                    + (mol[atom_j].coord(i) - mol[atom_i].coord(i))
+                    * average_bond_length(mol, conns, mol[atom_i].Z(),
+                    inputs.at("capping atom").value<atom_type>().Z())
+                    / original_bond;
+                }
```

Comment:
```suggestion
                 const auto cap_Z = new_cap.Z();
                 const auto Zi    = mol[atom_i].Z();
                 const auto r0 =average_bond_length(mol, conns, Zi, cap_Z);
                 for(atom_type::size_type i = 0; i < 3; ++i){
                    const auto qi = mol[atom_i].coord(i);
                    const auto dq = mol[atom_j].coord(i) - qi;
                    
                    new_cap.coord(i) = qi + dq * (r0 / original_bond);
                }
```
That was pretty hard to read. The general rule of thumb is not to let one semi-colon terminated line of code go over the 80 character mark. If it's going to go over 80 characters, try and declare intermediates like I did there.